pkg/dns/ibm/public/cis_provider.go

```diff
@@ -160,41 +166,45 @@ func (p *Provider) createOrUpdateDNSRecord(record *iov1.DNSRecord, zone configv1
 	// "." when it creates a wildcard DNS record.
 	dnsName := strings.TrimSuffix(record.Spec.DNSName, ".")
 	listOpt.SetName(dnsName)
-	for _, target := range record.Spec.Targets {
-		listOpt.SetContent(target)
-		result, response, err := dnsService.ListAllDnsRecords(listOpt)
+
+	result, response, err := dnsService.ListAllDnsRecords(listOpt)
+	if err != nil {
+		// Avoid continuing with an invalid list response, as we can't determine
+		// whether to create or update the DNS record, leading to further issues.
+		if response == nil || response.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("createOrUpdateDNSRecord: failed to list the dns record: %w", err)
+		}
+	}
+	if result == nil || result.Result == nil {
+		return fmt.Errorf("createOrUpdateDNSRecord: ListAllDnsRecords returned nil as result")
+	}
+
+	target := record.Spec.Targets[0]
+	if len(result.Result) == 0 {
+		createOpt := dnsService.NewCreateDnsRecordOptions()
+		createOpt.SetName(record.Spec.DNSName)
+		createOpt.SetType(string(record.Spec.RecordType))
+		createOpt.SetContent(target)
+		createOpt.SetTTL(record.Spec.RecordTTL)
+		_, _, err := dnsService.CreateDnsRecord(createOpt)
 		if err != nil {
-			if response != nil && response.StatusCode != http.StatusNotFound {
-				return fmt.Errorf("createOrUpdateDNSRecord: failed to list the dns record: %w", err)
-			}
-			continue
+			return fmt.Errorf("createOrUpdateDNSRecord: failed to create the dns record: %w", err)
 		}
-		if result == nil || result.Result == nil {
-			return fmt.Errorf("createOrUpdateDNSRecord: ListAllDnsRecords returned nil as result")
+		log.Info("created DNS record", "record", record.Spec, "zone", zone, "target", target)
+	} else {
+		if result.Result[0].ID == nil {
```

Comment:
If it's possible for `len(result.Result) > 1`, then you'd need to iterate over the results, wouldn't you?